### ISSUE TITLE
Enabling clustering in Redis fails with Redis Gear.

### ISSUE THREAD

**Issue by dallaybatta** [RedisGears/RedisGears#825]
I have build the redis gear from the 1.2 branch few weeks back and was able to run it with the redis from the master, the version of the redis is here

base) vicky@vicky-Latitude-5531:~/development/redis-git/redis$ git log -1
commit c66eaf4e4a019fe379556bd6f725524cca0e2d96 (HEAD -> unstable, origin/unstable, origin/HEAD)
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Wed Aug 31 10:13:48 2022 +0300

    Bump vmactions/freebsd-vm from 0.2.3 to 0.2.4 (#11203)
    
    Bumps [vmactions/freebsd-vm](https://github.com/vmactions/freebsd-vm) from 0.2.3 to 0.2.4.
    - [Release notes](https://github.com/vmactions/freebsd-vm/releases)
    - [Commits](https://github.com/vmactions/freebsd-vm/compare/v0.2.3...v0.2.4)
    
    ---
    updated-dependencies:
    - dependency-name: vmactions/freebsd-vm
      dependency-type: direct:production
      update-type: version-update:semver-patch
    ...

The redis was build from the master few weeks back. While trying to get the redis gear up with the cluster, the redis crashed with following error

***************************************************************************************************************
(base) vicky@vicky-Latitude-5531:~/development/redis-git/redis/src$ redis-server ../redis.conf 
55179:C 21 Sep 2022 17:58:00.541 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
55179:C 21 Sep 2022 17:58:00.541 # Redis version=255.255.255, bits=64, commit=c66eaf4e, modified=1, pid=55179, just started
55179:C 21 Sep 2022 17:58:00.541 # Configuration loaded
55179:M 21 Sep 2022 17:58:00.541 * Increased maximum number of open files to 10032 (it was originally set to 1024).
55179:M 21 Sep 2022 17:58:00.541 * monotonic clock: POSIX clock_gettime
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 255.255.255 (c66eaf4e/1) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                  
 (    '      ,       .-`  | `,    )     Running in cluster mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 55179
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           https://redis.io       
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               

55179:M 21 Sep 2022 17:58:00.542 * <rg> RedisGears version 1.2.6, git_sha=fdc116fce4daa84b784d6626eeaac544ed65ffea, compiled_os=linux-focal-x64

=== REDIS BUG REPORT START: Cut & paste starting from here ===
55179:M 21 Sep 2022 17:58:00.542 # Redis 255.255.255 crashed by signal: 11, si_code: 1
55179:M 21 Sep 2022 17:58:00.542 # Accessing address: (nil)
55179:M 21 Sep 2022 17:58:00.542 # Crashed running the instruction at: 0x55b03b033d14

------ STACK TRACE ------
EIP:
redis-server 127.0.0.1:6379 [cluster](RM_Call+0x674)[0x55b03b033d14]

Backtrace:
/lib/x86_64-linux-gnu/libpthread.so.0(+0x14420)[0x7f8289365420]
redis-server 127.0.0.1:6379 [cluster](RM_Call+0x674)[0x55b03b033d14]
/home/vicky/development/redis-git/RedisGears/redisgears.so(GearsGetRedisVersion+0x53)[0x7f82890bbf73]
/home/vicky/development/redis-git/RedisGears/redisgears.so(RedisGears_OnLoad+0x8a)[0x7f82890cce8a]
/home/vicky/development/redis-git/RedisGears/redisgears.so(RedisModule_OnLoad+0x1145)[0x7f82890d1a95]
redis-server 127.0.0.1:6379 [cluster](moduleLoad+0xc3)[0x55b03b035143]
redis-server 127.0.0.1:6379 [cluster](moduleLoadFromQueue+0x49)[0x55b03b035479]
redis-server 127.0.0.1:6379 [cluster](main+0x4ae)[0x55b03af8438e]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f8289183083]
redis-server 127.0.0.1:6379 [cluster](_start+0x2e)[0x55b03af8482e]

------ REGISTERS ------
55179:M 21 Sep 2022 17:58:00.543 # 
RAX:0000000000000000 RBX:00007f8288c331c0
RCX:00007ffd73c7f310 RDX:0000000000000000
RDI:0000000000000000 RSI:00007f8288cb4060
RBP:00007f8288cb6f00 RSP:00007ffd73c7fba0
R8 :0000000000000000 R9 :00007ffd73c7fbb4
R10:00007f82892e53d0 R11:00007f8288c63223
R12:0000000000020600 R13:0000000000000000
R14:0000000000000000 R15:0000000000000000
RIP:000055b03b033d14 EFL:0000000000010202
CSGSFS:002b000000000033
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fbaf) -> 00007f8288c9e0f0
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fbae) -> 00007ffd73c7fc60
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fbad) -> 000000000000706d
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fbac) -> 00000006b38a6c31
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fbab) -> 00007f8289105ee8
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fbaa) -> 0000000000000002
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba9) -> 000055b03af9e111
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba8) -> 00007f8288cb6c01
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba7) -> 5edf7f7d0d731f00
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba6) -> 00007ffd73c7fbe0
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba5) -> 00007ffd73c7fcd0
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba4) -> 0000003000000020
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba3) -> 00007ffd73c7fc20
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba2) -> 0000000000000000
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba1) -> 0000000200000002
55179:M 21 Sep 2022 17:58:00.543 # (00007ffd73c7fba0) -> 0000000000000030

------ INFO OUTPUT ------
Segmentation fault (core dumped)
(base) vicky@vicky-Latitude-5531:~/development/redis-git/redis/src$ 
***************************************************************************************************************

**Comment by MeirShpilraien** [comment]
You are running with Redis from unstable branch. Please try to run with official Redis version.

**Comment by dallaybatta** [comment]
> You are running with Redis from unstable branch. Please try to run with official Redis version.

Yes I guess so. I will try but not sure if I will have to build the RedisGear/RedisTimeSeries and RedisAI again. Let me try use the stable redis server.

**Comment by MeirShpilraien** [comment]
No need to rebuilt all the modules, only Redis.

**Comment by dallaybatta** [comment]
I had build Redis6.2, to be precise Redis 6.2.7 (e6f67092/0) 64 bit.
However I have some observations to share

I did build the server and it failed again with some error( cluster-enabled) while I did not bother to note the details of error, I then rebuild the gear module and enabled the clustering with it and it worked. However I did not build the AI and Timeseries module it worked after I removed the dumb file as I was getting "Can't handle RDB format version 10" error.

I have not noted the sequence and don't have the exact error of the redis when I ran without rebuilding the redisgear module.

After I read your latest message I again tested with the older redis gear module and to my surprise this time it worked, so I guess the issue that I was facing for the first time could be due to the RDB file mismatch version.

Thanks,
Vicky
